### PR TITLE
[Connection] Add auth_mode for AzureOpenAIConnection

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/azure/_models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,1993 @@
+# coding=utf-8
```
Comment (brynn-code) [inline src/promptflow/promptflow/azure/_models/__init__.py:1]: Discuss if we still depends on azure-ai-ml 1.14.0's model

File: src/promptflow/dev-connections.json.example [hunk 2]
```diff
@@ -9,6 +9,16 @@
     },
     "module": "promptflow.connections"
   },
+  "azure_open_ai_aad": {
+    "type": "AzureOpenAIConnection",
+    "value": {
+      "api_base": "aoai-api-endpoint",
+      "api_type": "azure",
+      "auth_mode": "meid_token",
```
Comment (brynn-code) [inline src/promptflow/dev-connections.json.example:17]: Discuss shall we add change log for auth_mode, or just waiting for tools ready and announce?

Comment (wangchao1230) [inline src/promptflow/dev-connections.json.example:17]: what does meid means?

File: src/promptflow/promptflow/azure/operations/_arm_connection_operations.py [hunk 3]
```diff
@@ -41,6 +44,11 @@ class ConnectionCategory:
     BingLLMSearch = "BingLLMSearch"
 
 
+class ConnectionAuthType:
+    ApiKey = "ApiKey"
+    AAD = "AAD"
```
Comment (wangchao1230) [inline src/promptflow/promptflow/azure/operations/_arm_connection_operations.py:49]: why AAD here and meid_token previously?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/azure/operations/_arm_connection_operations.py (hunk 3 above)
Comment (brynn-code) [reply src/promptflow/promptflow/azure/operations/_arm_connection_operations.py:49]: AAD here is one of the connection AuthType workspace RP returned to us, 'auth_mode': 'meid_token' is the customer facing value in connection yaml and entity.

**Review by crazygao — APPROVED** [review]

**Pull request merged**